uxarray/grid/coordinates.py

```diff
@@ -244,6 +322,211 @@ def _construct_face_centroids(node_x, node_y, node_z, face_nodes, n_nodes_per_fa
     return _normalize_xyz(centroid_x, centroid_y, centroid_z)
 
 
+def haversine_distance(point1, point2):
```

Comment:
please use the _angle_of_two_vector function to avoid the trigonometric functions, which might introduce great precision problem

Reply:
done, thanks!!